src/gui/DatabaseWidget.cpp

```diff
@@ -1980,8 +1980,10 @@ bool DatabaseWidget::saveAs()
 
     QString oldFilePath = m_db->filePath();
     if (!QFileInfo::exists(oldFilePath)) {
-        oldFilePath =
-            QDir::toNativeSeparators(config()->get(Config::LastDir).toString() + "/" + tr("Passwords").append(".kdbx"));
+        QVariant defaultFileName = config()->get(Config::DefaultDatabaseFileName);
+        oldFilePath = QDir::toNativeSeparators(config()->get(Config::LastDir).toString() + "/"
+                                               + (defaultFileName.isNull() ? tr("Passwords").append(".kdbx")
```

Comment:
Use `isEmpty()` instead, you want the config variable to contain at least one character.

Reply:
Yes, you're right, thanks! I've just pushed the suggested changes (as 018e270).